README.md

```diff
@@ -10,27 +10,38 @@
 
 **Status:** In development (not for production use)
 
 ## Purpose
 
 ### What it does
 
+#### The `lint` command
+
+The `lint` command scans your project for modifications that likely need to be made to be compatible with A3. The lint command will work well with basically all projects and detects many issues. Here are just a few examples of what the linter can detect:
+
+- The need to rename `lib/modules` to `modules`.
+- The need to change `{{ apos.area(...) }}` to `{% area ... %}`.
+- The need to move code from `construct()` to `methods()`, `handlers()`, etc.
+
+Since the linter is very tolerant it is a good candidate for use with nearly all A2 projects that are migrating to A3.
+
+#### The `upgrade` command
+
+The Code Upgrader also offers an `upgrade` command to directly upgrade your project. While this is a great feature, keep in mind that not every module and project is a good candidate for automated code upgrading as there are an infinite number of ways projects can be structured. The `upgrade` command works best on projects that adhere very closely to the coding style of the official A2 sample projects and documentation.
+
+There are also many needed changes that the upgrade command cannot handle on its own. So a successful upgrade will always involve reviewing the output of the `lint` command as well as the Apostrophe documentation.
+
 The code upgrader will convert Apostrophe 2 codebases for installable modules *and* full A2 websites so they are *mostly ready* to run Apostrophe 3. This includes:
 - Moving modules from `lib/modules` to the `modules` directory.
 - Renaming most project-level Apostrophe core module customization directories to the A3 equivalents.
-- Converting field schemas, columns, and similarly structured features to the A3 "cascade" configuration structure.
-- Converting utility methods such as `addHelper()`, `apiRoute()`, and others to A3 module customization functions.
-- Moving code in `beforeConstruct`, `construct`, and `afterConstruct` that can't otherwise be converted into appropriate A3 module functions.
-- Inlining the content of any files used in module configuration with the `require('./lib/something')(self, options)` pattern. This allows the tool to run the rest of its actions on that code as well. Those complete "required" files are removed.
+- Converting field schemas, columns, and similarly structured features to the A3 "cascade" configuration structure, if the existing module follows the structure of our A2 example projects closely enough.
+- Converting utility methods such as `addHelper()`, `apiRoute()`, and others to A3 module customization functions, again if the project closely follows the structure of our typical projects.
+- Moving code in `construct`, and `afterConstruct` that can't otherwise be converted into appropriate A3 module functions.
```

Comment:
I did see `beforeConstruct` material moved to `beforeSuperClass`. has that changed?